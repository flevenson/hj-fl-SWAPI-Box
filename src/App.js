import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import NavBar from './components/NavBar/NavBar.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      people: [],
      planets: [],
      vehicles: []
    }
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>SWAPI Box</h1>
          <NavBar />
        </header>
      </div>
    );
  }
}

export default App;
