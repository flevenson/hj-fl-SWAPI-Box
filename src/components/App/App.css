body {
  background-image: url('../../images/Background.png');
  background-repeat: no-repeat;
  background-position: right top;
  background-size: cover; 
}


.app {
  display: grid;
  grid-template-rows: 100px 1fr;
  grid-template-columns: 300px 1fr;
  height: 100vh;
}

.app-header {
  grid-row: 1 / span 1;
  display: grid;
  grid-column: 1 / span 2;
  /*background: blue;*/
}

.site-heading {
  font-family: 'Fugaz One', cursive;
  font-size: 4rem;
  margin: 25px 25px;
  color: #4A3934;
}

.app-aside {
  grid-column: 1 / span 1;
  grid-row: 2 / span 1;
  /*background: red;*/
}

.nav-bar-div{
  grid-column: 2 / span 1;
  display: flex;
  align-item: center;
  justify-content: flex-end;
}