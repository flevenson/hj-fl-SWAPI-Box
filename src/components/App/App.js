import React, { Component } from 'react';
import './App.css';
import NavBar from '../NavBar/NavBar.js';
import SideText from '../SideText/SideText.js';
import CardContainer from '../CardContainer/CardContainer.js';
import * as Cleaner from '../../helper.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      people: {},
      planets: {},
      vehicles: {},
      filmText: {},
      selected: '',
      display: {}
    }
  }

  async componentDidMount() {
    const starWarsApi = 'https://swapi.co/api/films/';
    const response = await fetch(starWarsApi);
    const data = await response.json();
    const filmText = await this.formatFilmText(data)
    this.setState({
      filmText: filmText
    })
    if (await this.getFromLocalStorage('display')) {
      const storedState = await this.getFromLocalStorage('display')
      await this.setState({ storedState })
    }
  }

  getData = async (buttonName) => {
    await this.setState({ selected: buttonName });
    await this.setState({ display: await Cleaner.fetchData(buttonName) });
    await this.setState({ [buttonName]: this.state.display});
    await this.addToLocalStorage(buttonName)
  }

  addToLocalStorage(buttonName) {
    localStorage.setItem(('display'), JSON.stringify(this.state[buttonName]))
    localStorage.setItem(('selected'), JSON.stringify(this.state.selected))
  }

  getFromLocalStorage = async (storedName) => {
    const storedDisplay = await JSON.parse(await localStorage.getItem('display'));
    const storedState = await JSON.parse(await localStorage.getItem('selected'));
    let localStoredState = {
      display: storedDisplay,
      selected: storedState
    }
    return localStoredState
  }

  handleNavClick = (event) => {
    let buttonName = event.target.getAttribute('name')
    this.getData(buttonName)
  }

  async formatFilmText(data) {
    const randomNumber = await Math.round(Math.random() * 6);
    const filmText = await data.results[randomNumber].opening_crawl
    const filmTitle = await data.results[randomNumber].title
    const episode = await data.results[randomNumber].episode_id
    const formattedFilm = await {
      text: filmText,
      title: filmTitle,
      episode: episode
    }  
    return formattedFilm
  }

  render() {
    const { people, selected, filmText, display, planets, vehicles } = this.state;

    return (
      <div className="app">
        <header className="app-header">
          <h1 className="site-heading">SWAPI Box</h1>
          <NavBar 
            getData={ this.getData } 
            selected={ selected } 
            handleNavClick={ this.handleNavClick }
          />
        </header>
        <aside className="app-aside">
          <SideText 
            filmText={ filmText } 
          />
        </aside>
        <main>
          <CardContainer 
            display={ display } 
            people={ people } 
            vehicles={ vehicles } 
            planets={ planets }
            selected={ selected }
          />
        </main>
      </div>
    );
  }
}

export default App;
