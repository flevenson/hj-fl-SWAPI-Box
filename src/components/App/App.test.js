import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';

describe('App', () => {
	let wrapper;
	let mountedWrapper;

	beforeEach(() => {
		wrapper = shallow(<App />)
		mountedWrapper = mount(<App />)
	})

	it('should match a snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('should call handleNavClick with the right params', () => {
		let	expected = {
			  type: 'click',
			  target: { getAttribute: jest.fn() }
			}

			mountedWrapper.find('.people').simulate('click')
			expect(mountedWrapper.instance().handleNavClick).toHaveBeenCalledWith(expected)
		})

	// it('should update the state of display when getData is called', () => {
	// 	let 
	// })
})
