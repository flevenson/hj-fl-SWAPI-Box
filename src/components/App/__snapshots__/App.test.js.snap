// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="app-header"
        >
          <h1
            className="site-heading"
          >
            SWAPI Box
          </h1>
          <NavBar
            getData={[Function]}
            handleNavClick={[Function]}
            selected=""
          />
        </header>,
        <aside
          className="app-aside"
        >
          <SideText
            filmText={Object {}}
          />
        </aside>,
        <main>
          <CardContainer
            display={Object {}}
            people={Object {}}
            planets={Object {}}
            selected=""
            vehicles={Object {}}
          />
        </main>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="site-heading"
            >
              SWAPI Box
            </h1>,
            <NavBar
              getData={[Function]}
              handleNavClick={[Function]}
              selected=""
            />,
          ],
          "className": "app-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SWAPI Box",
              "className": "site-heading",
            },
            "ref": null,
            "rendered": "SWAPI Box",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "getData": [Function],
              "handleNavClick": [Function],
              "selected": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <SideText
            filmText={Object {}}
          />,
          "className": "app-aside",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "filmText": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "aside",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CardContainer
            display={Object {}}
            people={Object {}}
            planets={Object {}}
            selected=""
            vehicles={Object {}}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "display": Object {},
            "people": Object {},
            "planets": Object {},
            "selected": "",
            "vehicles": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="app-header"
          >
            <h1
              className="site-heading"
            >
              SWAPI Box
            </h1>
            <NavBar
              getData={[Function]}
              handleNavClick={[Function]}
              selected=""
            />
          </header>,
          <aside
            className="app-aside"
          >
            <SideText
              filmText={Object {}}
            />
          </aside>,
          <main>
            <CardContainer
              display={Object {}}
              people={Object {}}
              planets={Object {}}
              selected=""
              vehicles={Object {}}
            />
          </main>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="site-heading"
              >
                SWAPI Box
              </h1>,
              <NavBar
                getData={[Function]}
                handleNavClick={[Function]}
                selected=""
              />,
            ],
            "className": "app-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SWAPI Box",
                "className": "site-heading",
              },
              "ref": null,
              "rendered": "SWAPI Box",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "getData": [Function],
                "handleNavClick": [Function],
                "selected": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SideText
              filmText={Object {}}
            />,
            "className": "app-aside",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "filmText": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "aside",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CardContainer
              display={Object {}}
              people={Object {}}
              planets={Object {}}
              selected=""
              vehicles={Object {}}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "display": Object {},
              "people": Object {},
              "planets": Object {},
              "selected": "",
              "vehicles": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
